src/bucket/atomic.rs

```diff
@@ -224,6 +232,17 @@ impl InputScope for AtomicBucket {
             .clone();
         InputMetric::new(move |value, _labels| scores.update(value))
     }
+
+    /// Observe a gauge value using a callback function.
+    fn observe(&self, name: &str, callback: GaugeCallback) {
+        let gauge = self.gauge(name);
+
+        self.inner
+            .write()
+            .expect("Aggregator")
+            .gauge_observers
+            .push(GaugeObserver { gauge, callback })
```

Comment:
Memory leak on periodic re-registration of the same metric, change vec to set or map indexed by full metric name. Existing instance is returned for gauge.